ant-protocol/src/storage/pointer.rs

```diff
@@ -0,0 +1,208 @@
+use crate::storage::{ChunkAddress, LinkedListAddress, PointerAddress, ScratchpadAddress};
+use bls::{Error as BlsError, PublicKey, SecretKey, Signature};
+use hex::FromHexError;
+use rand::thread_rng;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+use xor_name::XorName;
+
+#[derive(Error, Debug)]
+pub enum PointerError {
+    #[error("Failed to decode hex string: {0}")]
+    HexDecoding(#[from] FromHexError),
+    #[error("Failed to create public key: {0}")]
+    BlsError(#[from] BlsError),
+    #[error("Invalid public key bytes length")]
+    InvalidPublicKeyLength,
+    #[error("Invalid signature")]
+    InvalidSignature,
+    #[error("Serialization error: {0}")]
+    SerializationError(String),
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct Pointer {
+    owner: PublicKey,
+    counter: u32,
+    target: PointerTarget,
+    signature: Signature,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub enum PointerTarget {
+    ChunkAddress(ChunkAddress),
+    LinkedListAddress(LinkedListAddress),
+    PointerAddress(PointerAddress),
+    ScratchpadAddress(ScratchpadAddress),
+}
```

Comment:
I suppose we omitted registers here on purpose?

Reply:
Should this include DataMap, not just for files but arbitrary metadata?

Reply:
Agreed +1 @happybeing